convert2rhel/main.py

```diff
@@ -198,6 +194,13 @@ def post_ponr_conversion():
     pkghandler.replace_non_red_hat_packages()
     loggerinst.task("Convert: List remaining non-Red Hat packages")
     pkghandler.list_non_red_hat_pkgs_left()
+    
+    # if user modified /etc/yum.conf, then comment out the distroverpkg variable in yum.conf
+    output, return_code = utils.run_subprocess("rpm -V yum", False, False)
```

Comment:
We need to find a better way to detect that the `/etc/yum.conf` file has been modified by the user.

What I've found out, the `rpm -V yum` is not what would give us that information.
I was thinking that maybe `rpm -Vf /etc/yum.conf` would do, but not really.

I'll try to find that out. You can also try to search for the solution. Or maybe @pirat89 would know?

Reply:
I think, that this could be way: `rpm -Vf /etc/yum.cong | grep 5` or `rpm -V yum | grep 5.` According to this table in this [article](https://www.g-loaded.eu/2012/03/26/restore-original-configuration-files-from-rpm-packages/) and [documentation](http://ftp.rpm.org/max-rpm/s1-rpm-verify-output.html) the number 5 means, that the file was modified - checksum is different. So when grep finds 5, it returns the line and exit code is 0, otherwise 1. 

I hope, that the grep is standard part of CentOS and Oracle Linux.

What's your opinion?

Reply:
Then we would need another MR to implement pipes in run_subprocess (see https://stackoverflow.com/questions/9393425/python-how-to-execute-shell-commands-with-pipe-but-without-shell-true) . @bocekm do you know  how often we would need the pipe in shell commands? Would it worse the feature?

but I think particularly in grep case, you could solve it just with python filter (https://docs.python.org/3/library/functions.html#filter)

Reply:
We haven't had the need for the pipe yet. There's always been more pythonic approach at hand.

The `/etc/yum.conf` is applicable on systems derived from RHEL 6 and 7. On RHEL 8 we need to query `/etc/dnf/dnf.conf`.

@ZhukovGreen, I'd use regexp in this case since `run_subprocess` returns a multiline string, not a list of lines. So you'd first need to split it and then it starts to look convoluted (`list(filter(lambda line: "/etc/yum.conf" in line, output.split()))`).

@hosekadam, we should strive for keeping the `main.py` without any logic and "low level" code as much as possible. In the `main.py` we should be ideally just calling functions/methods defined in other files. So, I'd call there just the `redhatrelease.YumConf().patch()` and then the conf modification check should happen within the `patch()` method.

# You can add a method similar to this one to the `YumConf` class
```python
@staticmethod
def _is_modified()
    """Return true if the YUM/DNF configuration file has been modified by the user."""
    if pkgmanager.TYPE == 'yum':
        conf = "/etc/yum.conf"
    elif pkgmanager.TYPE == 'dnf':
        conf = "/etc/dnf/dnf.conf"

    output, _ = utils.run_subprocess("rpm -Vf %s" % conf, print_output=False)
    # rpm -Vf does not return information about the queried file but about all files owned by the rpm
    # that owns the queried file
    return True if re.search(r".{2}5.*? %s$" % conf, output, re.MULTILINE) else return False
```

Please add unit test as well. Thanks.

Reply:
@bocekm no better way, I would do something like you did in the last snippet :)